apps/nextjs-website/src/lib/fetch/types/PageSection.ts

```diff
@@ -360,6 +360,26 @@ const PageSwitchSectionCodec = t.strict({
   ),
 });
 
+const FramedVideoSectionCodec = t.strict({
+  __component: t.literal('sections.framed-video'),
+  sectionID: t.union([t.string, t.null]),
+  theme: t.union([t.literal('light'), t.literal('dark')]),
+  videoURL: t.union([t.string, t.null]),
+  video: t.union([VideoCodec, t.null]),
```

Comment:
```suggestion
  video: t.strict({
    data: t.union([
      t.strict({
        attributes: t.strict({ url: t.string }),
      }),
      t.null,
    ]),
  }),
```
You're misusing VideoCodec.
You only need the "src" part of it like the suggestion shows.